packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerToolbar.tsx

```diff
@@ -95,7 +92,7 @@ export const DateRangePickerToolbar = React.forwardRef(function DateRangePickerT
 
   return (
     <DateRangePickerToolbarRoot
-      toolbarTitle={toolbarTitle}
+      toolbarTitle={localeText.dateRangePickerDefaultToolbarTitle}
```

Comment:
IMHO—with this implementation the `Default` in these locales start to make less sense and from the initial look appears strange.

What do you think about leaving the more explicit implementation—but adding additional typing?

```diff
diff --git a/packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerToolbar.tsx b/packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerToolbar.tsx
index 3869d4754..179a715bc 100644
--- a/packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerToolbar.tsx
+++ b/packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerToolbar.tsx
@@ -11,6 +11,7 @@ import {
   useLocaleText,
   ExportedBaseToolbarProps,
 } from '@mui/x-date-pickers/internals';
+import { PickersInputComponentLocaleText } from '@mui/x-date-pickers';
 import { DateRange, CurrentlySelectingRangeEndProps } from '../internal/models';
 import {
   DateRangePickerToolbarClasses,
@@ -77,7 +78,8 @@ export const DateRangePickerToolbar = React.forwardRef(function DateRangePickerT
     toolbarFormat,
   } = props;
 
-  const localeText = useLocaleText();
+  const localeText = useLocaleText<TDate, Partial<PickersInputComponentLocaleText<TDate>>>();
+  const toolbarTitle = localeText.toolbarTitle ?? localeText.dateRangePickerDefaultToolbarTitle;
 
   const startDateValue = start
     ? utils.formatByString(start, toolbarFormat || utils.formats.shortDate)
@@ -92,7 +94,7 @@ export const DateRangePickerToolbar = React.forwardRef(function DateRangePickerT
 
   return (
     <DateRangePickerToolbarRoot
-      toolbarTitle={localeText.dateRangePickerDefaultToolbarTitle}
+      toolbarTitle={toolbarTitle}
       isMobileKeyboardViewOpen={isMobileKeyboardViewOpen}
       toggleMobileKeyboardView={toggleMobileKeyboardView}
       isLandscape={false}
diff --git a/packages/x-date-pickers-pro/src/DateRangePicker/shared.ts b/packages/x-date-pickers-pro/src/DateRangePicker/shared.ts
index ef3cb324c..90ef547d5 100644
--- a/packages/x-date-pickers-pro/src/DateRangePicker/shared.ts
+++ b/packages/x-date-pickers-pro/src/DateRangePicker/shared.ts
@@ -1,5 +1,4 @@
 import * as React from 'react';
-import { LocalizedComponent, PickersInputLocaleText } from '@mui/x-date-pickers';
 import {
   BasePickerProps,
   PickerStateValueManager,
@@ -72,13 +71,10 @@ export function useDateRangePickerDefaultizedProps<
 >(
   props: Props,
   name: string,
-): LocalizedComponent<
-  TDate,
-  DefaultizedProps<
-    Props,
-    'calendars' | keyof BaseDateValidationProps<TDate>,
-    { inputFormat: string }
-  >
+): DefaultizedProps<
+  Props,
+  'calendars' | keyof BaseDateValidationProps<TDate>,
+  { inputFormat: string }
 > {
   const utils = useUtils<TDate>();
   const defaultDates = useDefaultDates<TDate>();
@@ -90,16 +86,6 @@ export function useDateRangePickerDefaultizedProps<
     name,
   });
 
-  const localeText = React.useMemo<PickersInputLocaleText<TDate> | undefined>(() => {
-    if (themeProps.localeText?.toolbarTitle == null) {
-      return themeProps.localeText;
-    }
-
-    return {
-      ...themeProps.localeText,
-      dateRangePickerDefaultToolbarTitle: themeProps.localeText.toolbarTitle,
-    };
-  }, [themeProps.localeText]);
 
   return {
     disableFuture: false,
@@ -109,7 +95,6 @@ export function useDateRangePickerDefaultizedProps<
     ...themeProps,
     minDate: applyDefaultDate(utils, themeProps.minDate, defaultDates.minDate),
     maxDate: applyDefaultDate(utils, themeProps.maxDate, defaultDates.maxDate),
-    localeText,
     components: { Toolbar: DateRangePickerToolbar, ...themeProps.components },
   };
 }
diff --git a/packages/x-date-pickers/src/internals/hooks/useUtils.ts b/packages/x-date-pickers/src/internals/hooks/useUtils.ts
index 4ce3676f9..1f5a99d97 100644
--- a/packages/x-date-pickers/src/internals/hooks/useUtils.ts
+++ b/packages/x-date-pickers/src/internals/hooks/useUtils.ts
@@ -6,7 +6,10 @@ import {
 import { DEFAULT_LOCALE } from '../../locales/enUS';
 import { PickersLocaleText } from '../../locales/utils/pickersLocaleTextApi';
 
-export const useLocalizationContext = <TDate>() => {
+export const useLocalizationContext = <
+  TDate,
+  AdditionalLocaleText extends Record<string, any>,
+>() => {
   const localization = React.useContext(MuiPickersAdapterContext);
   if (localization === null) {
     throw new Error(
@@ -36,15 +39,18 @@ export const useLocalizationContext = <TDate>() => {
     ...localization,
     localeText,
   } as Omit<MuiPickersAdapterContextValue<TDate>, 'localeText'> & {
-    localeText: PickersLocaleText<TDate>;
+    localeText: PickersLocaleText<TDate> & AdditionalLocaleText;
   };
 };
 
-export const useUtils = <T>() => useLocalizationContext<T>().utils;
+export const useUtils = <T, L extends Record<string, any> = any>() =>
+  useLocalizationContext<T, L>().utils;
 
-export const useDefaultDates = <T>() => useLocalizationContext<T>().defaultDates;
+export const useDefaultDates = <T, L extends Record<string, any> = any>() =>
+  useLocalizationContext<T, L>().defaultDates;
 
-export const useLocaleText = <T>() => useLocalizationContext<T>().localeText;
+export const useLocaleText = <T, L extends Record<string, any> = any>() =>
+  useLocalizationContext<T, L>().localeText;
 
 export const useNow = <TDate>(): TDate => {
   const utils = useUtils<TDate>();
```

Reply:
> IMHO—with this implementation the Default in these locales start to make less sense and from the initial look appears strange.

I did not do it yet, but for me we should rename `datePickerDefaultToolbarTitle` into `datePickerToolbarTitle` (same for the rest).
It's not a `default` anymore now that we removed the `props.label` and `props.toolbarTitle` fallback.

____

The main problem I with your diff here is for custom toolbars.
If a developer create a custom toolbar for let's day the `DatePicker`, he will have to be aware of this fallback logic, instead of having it hidden inside the `useDatePickerDefaultizedProps`.

By the way, you removed the `LocalizedComponent` type, but when is the `toolbarTitle` translation key added ? 
Would we add it directly to `PickersComponentAgnosticLocaleText` ?
If so, would would it be the only nullable key from `PickersComponentAgnosticLocaleText` ? That's what I am trying to avoid.

Reply:
> The main problem I with your diff here is for custom toolbars...

I see your point—but on the other hand:
1. our current implementation already handles the `tooltipTitle` logic inside the component (whoever had custom toolbar implementations—wouldn't need to bother much as the implementation is similar, just with one case dropped (label));
2. whoever would be looking into creating a custom toolbar—would probably start with our component as base anyway;
___
> By the way, you removed the LocalizedComponent type, but when is the toolbarTitle translation key added ?

If I understand your question correctly—the following prop in `basePickerProps` handles it. 🤔 

```tsx
localeText?: PickersInputComponentLocaleText<TDate>;
```

Reply:
> our current implementation already handles the tooltipTitle logic inside the component (whoever had custom toolbar implementations—wouldn't need to bother much as the implementation is similar, just with one case dropped (label));

I feel like how easy it is to customize today should not be our baseline, we could improve. 

> If I understand your question correctly—the following prop in basePickerProps handles it. thinking

OK but where does the typing say that the `localeText` of `DatePicker` is not the same than the locale text of `LocalizationProvider` ?
`BasePickerProps` says its a `PickersInputComponentLocaleText` but `LocalizationProvider` expected a `PickersInputLocaleText`.

Reply:
For the custom toolbar argument, I think that going with an approach that still has the explicit fallback can make sense.
I would have liked to get rid of it but it's not a no-go for me.

If we can have a better typing approach, I'm fine with your global idea.

Reply:
As per discussion via Slack—I'm closing this discussion as the main problem I wanted to solve was the ambiguity in using the toolbar props with `Default` part in them—which we will ditch with a future PR.